### DISCUSSION TITLE
How can a consent and verifiable presentation use case be set up

### DISCUSSION THREAD

**Discussion by coolbung** [Sunbird-RC/community#256] [Ideas & Enhancements]
We wish to set up the below use cases based on Sunbird RC
1. Credential owner should be able to choose which credentials are available in their "public profile"
2. Say, the platform allows issuing multiple VC Proofs (education, experience) to a user. How can the user share selected proofs with a consumer (eg: only education experience to be shared, or only domain specific experience creds to be shared) ? This could happen online if the platform facilitates, or the user may be allowed to export a composite VC (or VP) that contains the chosen VCs.
3. The consumer can either be online, connected to the platform and can do a realtime verification. Or the consumer may be offline, and may need to verify using a VP. In case of offline verification, what are the recommendations in terms of caching the public keys of issuers ?

Is this best built as part of Sunbird RC Itself ? Or can the VP work being done by you @ChakshuGautam @gamemaker1 be leveraged for this ?

**Comment by sankarshanmukhopadhyay** [comment]
> We wish to set up the below use cases based on Sunbird RC
> 
>     1. Credential owner should be able to choose which credentials are available in their "public profile"

Do you mean the "holder" or "recipient" of the credentials? Ownership is not a concept that is immediately used in context of the flow.

>     2. Say, the platform allows issuing multiple VC Proofs (education, experience) to a user. How can the user share selected proofs with a consumer (eg: only education experience to be shared, or only domain specific experience creds to be shared) ? This could happen online if the platform facilitates, or the user may be allowed to export a composite VC (or VP) that contains the chosen VCs.

Or, could also be done through VCs stored in a digital wallet and then composited into a presentation. See https://w3c-ccg.github.io/vp-request-spec/#usage-scenarios as an example

>     3. The consumer can either be online, connected to the platform and can do a realtime verification. Or the consumer may be offline, and may need to verify using a VP. In case of offline verification, what are the recommendations in terms of caching the public keys of issuers ?

The offline scenario - is this designed to be fully offline?
